### PR TITLE
add DeformationFieldTransform

### PR DESCRIPTION
**Pull request by bogovicj**
Among other things, this has enabled me to wrap warp fields from [ANTs](http://stnava.github.io/ANTsR/) as a RealTransform.

### CODE REVIEW

File: src/main/java/net/imglib2/realtransform/DeformationFieldTransform.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+public class DeformationFieldTransform<T extends RealType<T>> implements RealTransform
+{
+
+	private final RealRandomAccessible< T > defFieldReal;
+
+	private final int numDim;
+
+	public DeformationFieldTransform( RandomAccessibleInterval< T > def )
+	{
+        this( Views.interpolate( def, new NLinearInterpolatorFactory< T >() ));
```
Comment (axtimwalde) [inline src/main/java/net/imglib2/realtransform/DeformationFieldTransform.java]: Indentation problem?  Please run autoformat with imglib2 style.

File: src/main/java/net/imglib2/realtransform/DeformationFieldTransform.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	@Override
+	public int numTargetDimensions()
+	{
+		return numDim;
+	}
+
+	@Override
+	public void apply( double[] source, double[] target )
+	{
+		RealRandomAccess< T > defFieldAccess = defFieldReal.realRandomAccess();
```
Comment (axtimwalde) [inline src/main/java/net/imglib2/realtransform/DeformationFieldTransform.java]: That creates a new `RealRandomAccess` per coordinate transfer which I believe will have significant runtime cost.  I suggest to have one `RealRandomAccess` per transform.  `RealTransform`s are not expected to be used by multiple threads in parallel.

File: src/main/java/net/imglib2/realtransform/DeformationFieldTransform.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+		{
+			target[ d ] += defFieldAccess.get().getRealDouble();
+			defFieldAccess.fwd( numDim );
+		}
+	}
+
+	@Override
+	public void apply( float[] source, float[] target )
+	{
+		RealRandomAccess< T > defFieldAccess = defFieldReal.realRandomAccess();
```
Comment (axtimwalde) [inline src/main/java/net/imglib2/realtransform/DeformationFieldTransform.java]: See above.

File: src/main/java/net/imglib2/realtransform/DeformationFieldTransform.java [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+		{
+			target[ d ] += defFieldAccess.get().getRealDouble();
+			defFieldAccess.fwd( numDim );
+		}
+	}
+
+	@Override
+	public void apply( RealLocalizable source, RealPositionable target )
+	{
+		RealRandomAccess< T > defFieldAccess = defFieldReal.realRandomAccess();
```
Comment (axtimwalde) [inline src/main/java/net/imglib2/realtransform/DeformationFieldTransform.java]: See above.

File: src/main/java/net/imglib2/realtransform/DeformationFieldTransform.java (hunk 2 above)
Comment (bogovicj) [inline src/main/java/net/imglib2/realtransform/DeformationFieldTransform.java]: Thanks, good catch.  Will make this change

**Pull request merged**